src/modules/dashboard/data/privilege-limits.ts

```diff
@@ -0,0 +1,43 @@
+import { Roles } from '@/global/enums/roles.enum';
+import { DashboardModulesEnum } from '../enums/dashboard-modules.enum';
+import { Privileges as PrivilegesEnum } from '@/global/enums/privileges.enum';
+
+type RolePrivilegeLimit = {
+  [role in Roles]: {
+    [module in DashboardModulesEnum]?: PrivilegesEnum[];
+  };
+};
+
+// ! IMPORTANT -> this should be moved to the backend and fetched from api
+export const rolePrivelegeLimits: RolePrivilegeLimit = {
+  [Roles.ADMIN]: {}, // admin always has full access to all modules
+  [Roles.MANAGER]: {
+    [DashboardModulesEnum.PRODUCTS]: [
+      PrivilegesEnum.MANAGE,
+      PrivilegesEnum.READONLY,
+    ],
+    [DashboardModulesEnum.STAFF_MEMBERS]: [
+      PrivilegesEnum.MANAGE,
+      PrivilegesEnum.READONLY,
+      PrivilegesEnum.NOT_ALLOWED,
+    ],
+    [DashboardModulesEnum.CATEGORIES]: [
+      PrivilegesEnum.MANAGE,
+      PrivilegesEnum.READONLY,
+    ],
+  },
+  [Roles.ASSISTANT]: {
+    [DashboardModulesEnum.PRODUCTS]: [
+      PrivilegesEnum.MANAGE,
+      PrivilegesEnum.READONLY,
+      PrivilegesEnum.NOT_ALLOWED,
+    ],
+    [DashboardModulesEnum.CATEGORIES]: [
+      PrivilegesEnum.READONLY,
+      PrivilegesEnum.NOT_ALLOWED,
+    ],
+    [DashboardModulesEnum.STAFF_MEMBERS]: [
+      PrivilegesEnum.NOT_ALLOWED,
+    ],
+  },
+};
```

Comment:
Tak się zastanawiam, nie byłoby łatwiej gdybyśmy ustawiali dla poszczególnych ról i modułów przywilej o podanej wartości?
Admin może wszystko = 3
Assistant w productsach => może readonly = 1
Assistant w staff membersach => not allowed = 0
Manager w staff membersach => manage = 2

`const privilegeHierarchy: Record<PrivilegesEnum, number> = {
    [PrivilegesEnum.FULL]: 3,
    [PrivilegesEnum.MANAGE]: 2,
    [PrivilegesEnum.READONLY]: 1,
    [PrivilegesEnum.NOT_ALLOWED]: 0,
  };`

Reply:
Not allowed mogło by być -1, aby uniknąć dodatkowych ifów w przyszłości, po prostu żaden przywilej usera się temu nie równa, więc nie powinien wgl widzieć tych linków/podstron